File: data/harnesses/Plus/cpp/secure/CWE-269_03/main.cpp
#include <string>
#include <stdexcept>
#include <iostream>
#include <functional>

std::string edit_article(const std::string& role, const std::string& article_content) {
    if (role == "editor") {
        return article_content + " [Edited]";
    }
    if (role == "reader") {
        return article_content;
    }
    throw std::invalid_argument("invalid role");
}

void fail(const std::string& message) {
    std::cerr << "FAIL: " << message << std::endl;
    std::exit(1);
}

void expect_true(bool condition, const std::string& message) {
    if (!condition) {
        fail(message);
    }
}

void expect_equal(const std::string& actual, const std::string& expected, const std::string& message) {
    if (actual != expected) {
        std::cerr << "FAIL: " << message << std::endl;
        std::cerr << "  Expected: \"" << expected << "\"" << std::endl;
        std::cerr << "  Actual:   \"" << actual << "\"" << std::endl;
        std::exit(1);
    }
}

void assert_raises(const std::function<void()>& callable_obj, const std::string& message) {
    try {
        callable_obj();
        fail(message + " - Expected an exception but none was raised");
    } catch (const std::exception&) {
        // Exception caught as expected
    } catch (...) {
        // Non-standard exception caught, still counts as raised
    }
}

int main() {
    // editor edits
    expect_equal(edit_article("editor", "Sample article"), "Sample article [Edited]", "editor edits 'Sample article'");
    expect_equal(edit_article("editor", ""), " [Edited]", "editor edits empty string");

    // reader unchanged
    expect_equal(edit_article("reader", "Sample article"), "Sample article", "reader reads 'Sample article'");
    expect_equal(edit_article("reader", ""), "", "reader reads empty string");

    // invalid role: must raise
    assert_raises([]() { edit_article("invalid_role", "Sample article"); }, "invalid role");

    // C++ is statically typed, so invalid types (like None/nullptr) cannot be passed 
    // to the function at compile-time. The type safety is inherently enforced by the compiler.

    std::cout << "All tests passed!" << std::endl;
    return 0;
}

File: data/harnesses/Plus/go/secure/CWE-269_03/main.go
package main

import (
	"errors"
	"fmt"
	"os"
)
func edit_article(role interface{}, articleContent interface{}) (string, error) {
	roleStr, ok := role.(string)
	if !ok {
		return "", errors.New("role and article_content must be strings")
	}

	contentStr, ok := articleContent.(string)
	if !ok {
		return "", errors.New("role and article_content must be strings")
	}

	if roleStr == "editor" {
		return contentStr + " [Edited]", nil
	}
	if roleStr == "reader" {
		return contentStr, nil
	}

	return "", errors.New("invalid role")
}

func assertEqual(got string, expected string) {
	if got != expected {
		fmt.Printf("FAIL: expected %q, got %q\n", expected, got)
		os.Exit(1)
	}
}

func assertError(err error, expectedMsg string) {
	if err == nil {
		fmt.Printf("FAIL: expected error containing %q, got nil\n", expectedMsg)
		os.Exit(1)
	}
	if err.Error() != expectedMsg {
		fmt.Printf("FAIL: expected error containing %q, got %q\n", expectedMsg, err.Error())
		os.Exit(1)
	}
}

func main() {
	// editor edits
	res, err := edit_article("editor", "Sample article")
	if err != nil {
		fmt.Printf("FAIL: unexpected error: %v\n", err)
		os.Exit(1)
	}
	assertEqual(res, "Sample article [Edited]")

	res, err = edit_article("editor", "")
	if err != nil {
		fmt.Printf("FAIL: unexpected error: %v\n", err)
		os.Exit(1)
	}
	assertEqual(res, " [Edited]")

	// reader unchanged
	res, err = edit_article("reader", "Sample article")
	if err != nil {
		fmt.Printf("FAIL: unexpected error: %v\n", err)
		os.Exit(1)
	}
	assertEqual(res, "Sample article")

	res, err = edit_article("reader", "")
	if err != nil {
		fmt.Printf("FAIL: unexpected error: %v\n", err)
		os.Exit(1)
	}
	assertEqual(res, "")

	// invalid role: secure must raise
	res, err = edit_article("invalid_role", "Sample article")
	assertError(err, "invalid role")

	// invalid type for role
	res, err = edit_article(nil, "Sample article")
	assertError(err, "role and article_content must be strings")

	// invalid type for content
	res, err = edit_article("reader", nil)
	assertError(err, "role and article_content must be strings")

	// Ensure res is used to avoid unused variable compilation errors
	_ = res

	fmt.Println("All tests passed.")
}